src/Symfony/Component/Validator/Constraints/Ulid.php

```diff
@@ -35,20 +35,27 @@ class Ulid extends Constraint
         self::TOO_LARGE_ERROR => 'TOO_LARGE_ERROR',
     ];
 
+    public const FORMAT_BASE_32 = 'base32';
+    public const FORMAT_BASE_58 = 'base58';
+
     public string $message = 'This is not a valid ULID.';
+    public string $format = self::FORMAT_BASE_32;
 
     /**
      * @param array<string,mixed>|null $options
      * @param string[]|null            $groups
+     * @param self::FORMAT_*|null      $format
      */
     public function __construct(
         ?array $options = null,
         ?string $message = null,
         ?array $groups = null,
         mixed $payload = null,
+        ?string $format = null,
     ) {
         parent::__construct($options, $groups, $payload);
 
         $this->message = $message ?? $this->message;
+        $this->format = $format ?? $this->format;
```

Comment:
we should also validate that the selected format is valid

Reply:
I did validate in the validator but indeed it may be a little late. Better to do it in the constructor 👍